traitsui/tests/ui_editors/test_data_frame_editor.py

```diff
@@ -6,16 +6,15 @@
 #  under the conditions described in the aforementioned license.  The license
 #  is also available online at http://www.enthought.com/licenses/BSD.txt
 
-import nose
+import unittest
 import numpy as np
 from numpy.testing import assert_array_equal
 
 
 try:
     from pandas import DataFrame
-except ImportError as exc:
-    print("Can't import Pandas: skipping")
-    raise nose.SkipTest
+except ImportError:
+    raise unittest.SkipTest("Can't import Pandas: skipping")
```

Comment:
Request change: Same as above, this prevents the module from being run without `discover`. Could we add something like `skip_if_no_pandas` or `requires_pandas` decorator in the testing tools module instead?

Reply:
Unlike the instance above, I was able to run the module and see all the tests skipped if I did this:
```

try:
    from pandas import DataFrame
except ImportError:
    _HAS_PANDAS = False
else:
    _HAS_PANDAS = True


@unittest.skipIf(not _HAS_PANDAS, "pandas DataFrame is not available.")
class TestDataFrameEditor(unittest.TestCase):
...

```

```
$ python -m unittest -v traitsui.tests.ui_editors.test_data_frame_editor
test_adapter_delete_end (traitsui.tests.ui_editors.test_data_frame_editor.TestDataFrameEditor) ... skipped 'pandas DataFrame is not available.'
test_adapter_delete_end_numerical_index (traitsui.tests.ui_editors.test_data_frame_editor.TestDataFrameEditor) ... skipped 'pandas DataFrame is not available.'
...
```

Reply:
Not raising is consistent with the behaviour of `nose`:
On master:
```
$ python -m nose.core -v traitsui.tests.ui_editors.test_data_frame_editor
/Users/kchoi/.edm/envs/traitsui-test-3.6-null/lib/python3.6/runpy.py:125: RuntimeWarning: 'nose.core' found in sys.modules after import of package 'nose', but prior to execution of 'nose.core'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
Failure: SkipTest () ... SKIP

----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (SKIP=1)
```

Hence I think this is within scope to bring the behaviour aligned with the existing one. That said, I agree this only brings a slight annoyance for the developers, and it is also not hard to change.

Reply:
I don't think it's equivalent because nose shows the whole module as one skipped test while unittest shows individual skipped tests. And total number of tests would also be different when running full test suites. So I would be inclined to leave it as is for now, and if it does cause issues, it can be changed later.

Reply:
OK, for later.